src/staking.rs

```diff
@@ -623,26 +657,60 @@ impl Module for StakeKeeper {
                 Ok(AppResponse::default())
             }
             StakingSudo::ProcessQueue {} => {
+                let staking_storage = prefixed_read(storage, NAMESPACE_STAKING);
+                let mut unbonding_queue = UNBONDING_QUEUE
+                    .may_load(&staking_storage)?
+                    .unwrap_or_default();
                 loop {
                     let mut staking_storage = prefixed(storage, NAMESPACE_STAKING);
-                    let front = UNBONDING_QUEUE.front(&staking_storage)?;
-                    match front {
+                    match unbonding_queue.front() {
                         // assuming the queue is sorted by payout_at
-                        Some((_, payout_at, _)) if payout_at <= block.time => {
+                        Some(Unbonding { payout_at, .. }) if payout_at <= &block.time => {
                             // remove from queue
-                            let (delegator, _, amount) =
-                                UNBONDING_QUEUE.pop_front(&mut staking_storage)?.unwrap();
+                            let Unbonding {
+                                delegator,
+                                validator,
+                                amount,
+                                ..
+                            } = unbonding_queue.pop_front().unwrap();
+
+                            // remove staking entry if it is empty
+                            let delegation = self
+                                .get_stake(&staking_storage, &delegator, &validator)?
+                                .map(|mut stake| {
+                                    // add unbonding amounts
+                                    stake.amount += unbonding_queue
+                                        .iter()
+                                        .filter(|u| {
+                                            u.delegator == delegator && u.validator == validator
+                                        })
+                                        .map(|u| u.amount)
+                                        .sum::<Uint128>();
+                                    stake
+                                });
+                            match delegation {
+                                Some(delegation) if delegation.amount.is_zero() => {
+                                    STAKES.remove(&mut staking_storage, (&delegator, &validator));
+                                }
+                                None => {
+                                    STAKES.remove(&mut staking_storage, (&delegator, &validator))
+                                }
+                                _ => {}
+                            }
 
                             let staking_info = Self::get_staking_info(&staking_storage)?;
-                            if amount > 0 {
+                            if !amount.is_zero() {
```

Comment:
Yes, it's the same. `amount` now is `Uint128` instead of `u128` and I don't like the look of `amount > Uint128::zero()` when there is an explicit message to check that.